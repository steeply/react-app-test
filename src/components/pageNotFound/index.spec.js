/*eslint-disable */

import React from 'react';
import { shallow } from 'enzyme';
import PageNotFound from './index';

describe('PageNotFound container', () => {
  let container;
  beforeEach(() => {
    container = shallow(<PageNotFound />);
  });

  it('+++ render the PageNotFound component', () => {
    expect(container.length).toEqual(1);
  });

  it('+++ is render tag <h1>', () => {
    expect(container.find('h1').text()).toEqual('Error - 404');
  });

  it('+++ check Snapshot', () => {
    expect(container).toMatchSnapshot();
  });

});

/*eslint-enable */