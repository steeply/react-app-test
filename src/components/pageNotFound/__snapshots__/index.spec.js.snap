// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound container +++ check Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageNotFound />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Error - 404
        </h1>,
        <h3>
          Page not found!
        </h3>,
      ],
      "className": "pageNotFound",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Error - 404",
        },
        "ref": null,
        "rendered": "Error - 404",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Page not found!",
        },
        "ref": null,
        "rendered": "Page not found!",
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Error - 404
          </h1>,
          <h3>
            Page not found!
          </h3>,
        ],
        "className": "pageNotFound",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Error - 404",
          },
          "ref": null,
          "rendered": "Error - 404",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Page not found!",
          },
          "ref": null,
          "rendered": "Page not found!",
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
